src/main/java/com/brandon3055/draconicevolution/api/modules/data/AutoFireData.java

```diff
@@ -0,0 +1,30 @@
+package com.brandon3055.draconicevolution.api.modules.data;
+
+import com.brandon3055.draconicevolution.api.modules.lib.ModuleContext;
+import net.minecraft.util.text.ITextComponent;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+
+public class AutoFireData implements ModuleData<AutoFireData> {
```

Comment:
You don't need this. Module data is for modules that do things like boosting a tool's stats by some amount. For example, `SpeedData` is used to store and combine the speed bonus from multiple modules. 
For your module, you just want the `NoData` type.